example/misc/infinite_set.cpp

```diff
@@ -56,6 +62,22 @@ namespace boost { namespace hana {
         }
     };
 
+    template <>
+    struct difference_impl<infinite_set_tag> {
```

Comment:
@ldionne 
This actually doesn't seem good to me logically. Although infinite set had union function implemented, we can't call it, because apparantly, union requires the class to be SetTheoretic meaning it requires minimal definition - all 3 functions union, intersection & difference. But, mathematically, it seems accurate though.
Any thoughts ?